app/services/profiles/update.rb

```diff
@@ -35,8 +48,8 @@ def call
       self
     end
 
-    def success?
-      @success
+    def update_user
+      @success = true if @user.update(@updated_user_attributes.to_h)
```

Comment:
This won't work, need to take into account the success of both updates on the profile sync and on the user model directly. Was busy tackling this.

Reply:
Should we update the user at all if the profile update didn't succeed? While they live on different models, the update process should be atomic and either succeed for both or neither. When doing a normal update everything gets rolled back when one field is invalid, I think the same should be true here too. 

We can potentially still do the following though:
1. assign the attributes
2. run validations
3. collect the errors to return to the user

Does this make sense? If not I can work on it on Monday.

Reply:
I tried something @citizen428, maybe you can have a look and correct where necessary or provide suggestions? Even though i added a flash for the error and success and its passing through the variables correctly, I couldn't get the flash to show up on the page, maybe coz i'm rendering through the users controller and updating through the profiles controller. I dunno, I'm looking into it and learning more.

Reply:
Re the flash: you're setting it, but not rendering anything after (this is a patch request). This should follow the usual Rails redirect on success and re-render on error idiom:

```ruby
    if update_result.success?
      flash[:success] = "Your profile has been updated"
      redirect_to ...
    else
      flash[:error] = "Error: #{update_result.error_message}"
      render ...
    end
```